src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -105,7 +106,7 @@ public GoogleBucketResource getOrCreateBucketForFile(String possibleBucketName,
         }
 
         return bucketService.getOrCreateBucket(bucketName, projectResource,
-            dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.BUCKET),
+            (GoogleRegion) dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.BUCKET),
```

Comment:
Unlike the case above, this one ends up needing a `GoogleRegion`.

Reply:
Right, because this is a google specific API

Reply:
The more specific type is only needed if the contract is different. If you moved the google contracts to the base class then you could use the base class type everywhere. The code would throw at runtime if it had the wrong typed object, which is the equivalent of getting a type mismatch exception here.